### ISSUE TITLE
[BUG] Ukelele breaks after hitting an enemy with it

### ISSUE THREAD

**Issue by Lunxara** [WeatherElectric/Lacrimosum#5]
I whacked an enemy with the Ukelele and then it broke and became completely unusable :(

[10:46:37.7304942] [Error  : Unity Log] IndexOutOfRangeException: Index was outside the bounds of the array.
Stack trace:
RoundManager.PlayRandomClip (UnityEngine.AudioSource audioSource, UnityEngine.AudioClip[] clipsArray, System.Boolean randomize, System.Single oneShotVolume, System.Int32 audibleNoiseID, System.Int32 maxIndex) (at <066a7b7eb8c24d4facd1e255e669803f>:IL_0029)
Lacrimosum.ItemScripts.Ukulele.HitShovel (System.Boolean cancel) (at <4f6e24dc6b1e49718ecbd5074ab31fa1>:IL_0334)
Lacrimosum.ItemScripts.Ukulele+<ReelUpShovel>d__15.MoveNext () (at <4f6e24dc6b1e49718ecbd5074ab31fa1>:IL_0196)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <e27997765c1848b09d8073e5d642717a>:IL_0026)

[10:46:37.7324959] [Error  :UnityDebuggerAssistant] 

--- Exception Handler ---

Exception Caught: System.IndexOutOfRangeException
Assembly: Assembly-CSharp
Message: Index was outside the bounds of the array.
Source: Assembly-CSharp

  --- Begin Frames ---

  --FRAME 1:
    In Assembly: Assembly-CSharp
    Target Method: RoundManager.PlayRandomClip

  --FRAME 2:
    In Assembly: Lacrimosum
    Target Method: Ukulele.HitShovel

  --FRAME 3:
    In Assembly: Lacrimosum
    Target Method: <ReelUpShovel>d__15.MoveNext

  --FRAME 4:
    In Assembly: UnityEngine.CoreModule
    Target Method: SetupCoroutine.InvokeMoveNext

  --- End Frames ---

--- End Exception Handler ---

**Comment by FragileDeviations** [comment]
This seems to be basegame? Fault lies in RoundManagar.PlayRandomClip

**Comment by Lunxara** [comment]
> This seems to be basegame? Fault lies in RoundManagar.PlayRandomClip

I don't believe so, as I've not had this issue with vanilla shovels or custom shovels from mods like VoidLeak, ImmersiveScrap, and CodeRebirth. I just imagine something is setup improperly. I think Xu said they're gonna see why it's happening and help you out ^^

**Comment by FragileDeviations** [comment]
> > This seems to be basegame? Fault lies in RoundManagar.PlayRandomClip
> 
> I don't believe so, as I've not had this issue with vanilla shovels or custom shovels from mods like VoidLeak, ImmersiveScrap, and CodeRebirth. I just imagine something is setup improperly. I think Xu said they're gonna see why it's happening and help you out ^^

VoidLeak uses the original unmodified shovel script, and I know since that's also my mod lol
But yea that'd be appreciated if they're gonna see why, I have no idea where to start on fixing it because I'm just using the vanilla shovel's code with an extra method call slapped in

**Comment by Lunxara** [comment]
> > > This seems to be basegame? Fault lies in RoundManagar.PlayRandomClip
> > 
> > 
> > I don't believe so, as I've not had this issue with vanilla shovels or custom shovels from mods like VoidLeak, ImmersiveScrap, and CodeRebirth. I just imagine something is setup improperly. I think Xu said they're gonna see why it's happening and help you out ^^
> 
> VoidLeak uses the original unmodified shovel script, and I know since that's also my mod lol But yea that'd be appreciated if they're gonna see why, I have no idea where to start on fixing it because I'm just using the vanilla shovel's code with an extra method call slapped in

Yeahh I just wonder why the additional method call is causing issues, maybe for now you could just try using the unmodified vanilla shovel code until Xu can help you look into it. Also I would recommend adding an option to not replace the menu music with ROR2's theme at all since I noticed that's rng, just to avoid potential DMCA issues for streams

**Comment by FragileDeviations** [comment]
> > > > This seems to be basegame? Fault lies in RoundManagar.PlayRandomClip
> > > 
> > > 
> > > I don't believe so, as I've not had this issue with vanilla shovels or custom shovels from mods like VoidLeak, ImmersiveScrap, and CodeRebirth. I just imagine something is setup improperly. I think Xu said they're gonna see why it's happening and help you out ^^
> > 
> > 
> > VoidLeak uses the original unmodified shovel script, and I know since that's also my mod lol But yea that'd be appreciated if they're gonna see why, I have no idea where to start on fixing it because I'm just using the vanilla shovel's code with an extra method call slapped in
> 
> Yeahh I just wonder why the additional method call is causing issues, maybe for now you could just try using the unmodified vanilla shovel code until Xu can help you look into it. Also I would recommend adding an option to not replace the menu music with ROR2's theme at all since I noticed that's rng, just to avoid potential DMCA issues for streams

Luckily, RoR2 music doesn't get copystriked at all. Chris is cool with licensing. If it was a problem, every ror2 stream ever would no longer exist.

**Comment by Lunxara** [comment]
> > > > > This seems to be basegame? Fault lies in RoundManagar.PlayRandomClip
> > > > 
> > > > 
> > > > I don't believe so, as I've not had this issue with vanilla shovels or custom shovels from mods like VoidLeak, ImmersiveScrap, and CodeRebirth. I just imagine something is setup improperly. I think Xu said they're gonna see why it's happening and help you out ^^
> > > 
> > > 
> > > VoidLeak uses the original unmodified shovel script, and I know since that's also my mod lol But yea that'd be appreciated if they're gonna see why, I have no idea where to start on fixing it because I'm just using the vanilla shovel's code with an extra method call slapped in
> > 
> > 
> > Yeahh I just wonder why the additional method call is causing issues, maybe for now you could just try using the unmodified vanilla shovel code until Xu can help you look into it. Also I would recommend adding an option to not replace the menu music with ROR2's theme at all since I noticed that's rng, just to avoid potential DMCA issues for streams
> 
> Luckily, RoR2 music doesn't get copystriked at all. Chris is cool with licensing. If it was a problem, every ror2 stream ever would no longer exist.

Hooray, my main concern though was that I figured he might not like his music used in other games but if he's chill with it then good

**Comment by FragileDeviations** [comment]
he's never taken down things that use his ror songs that arent getting any money made off of them and are directly ror2 based, the mod's ror2 based and i dont make money off it

**Comment by Lunxara** [comment]
> he's never taken down things that use his ror songs that arent getting any money made off of them and are directly ror2 based, the mod's ror2 based and i dont make money off it

Yay :D

**Comment by DSMister** [comment]
Hi, piling on the issue here.

It seems that I can only use the ukelele once before it's completely unusable. It doesn't seem to be directly affected by hitting enemies, either.

**Comment by FragileDeviations** [comment]
That is what the entire issue is about lol

**Comment by DSMister** [comment]
Ah, sorry. I thought the initial report was about it only being affected by swinging the ukelele at enemies.
